frontend/src/employee-frontend/components/holiday-term-periods/TermsSection.tsx

```diff
@@ -26,12 +34,63 @@ export default React.memo(function TermsSection() {
   const { i18n } = useTranslation()
   const navigate = useNavigate()
 
+  const [termToEdit, setTermToEdit] = useState<UUID>()
+  const [showEditModal, setShowEditModal] = useState(false)
+
+  const [termToDelete, setTermToDelete] = useState<UUID>()
+  const [showDeletionModal, setShowDeletionModal] = useState(false)
+
   const preschoolTerms = useQueryResult(preschoolTermsQuery())
 
+  const { mutateAsync: deletePreschoolTerm } = useMutationResult(
+    deletePreschoolTermMutation
+  )
+
+  const onDeletePreschoolTerm = useCallback(
+    (termId: UUID) => {
+      setTermToDelete(termId)
+      setShowDeletionModal(true)
+    },
+    [setTermToDelete, setShowDeletionModal]
+  )
+
+  const deletePreschoolTermHandle = useCallback(
+    () => (termToDelete ? deletePreschoolTerm(termToDelete) : Promise.reject()),
+    [termToDelete, deletePreschoolTerm]
+  )
+
+  const onCloseDeletionModal = useCallback(() => {
+    setTermToDelete(undefined)
+    setShowDeletionModal(false)
+  }, [setTermToDelete, setShowDeletionModal])
+
+  const onEditTermHandle = useCallback(
+    (term: PreschoolTerm) => {
+      if (term.finnishPreschool.start.isBefore(LocalDate.todayInSystemTz())) {
+        setTermToEdit(term.id)
+        setShowEditModal(true)
+      } else {
+        navigate(`/holiday-periods/preschool-term/${term.id}`)
+      }
+    },
+    [setShowEditModal, navigate]
+  )
+
+  const onCloseEditModal = useCallback(() => {
+    setTermToEdit(undefined)
+    setShowEditModal(false)
+  }, [setShowEditModal, setTermToEdit])
+
   const navigateToNewTerm = useCallback(() => {
     navigate('/holiday-periods/preschool-term/new')
   }, [navigate])
 
+  const closeModalAndNavigateToEditTerm = useCallback(() => {
+    if (termToEdit) {
+      navigate(`/holiday-periods/preschool-term/${termToEdit}`)
+    }
+  }, [navigate, termToEdit])
+
```

Comment:
On line 97 I'd consider using `AddButtonRow` instead of `AddButton` to keep the alignment more coherent with the other sections on the same page

Reply:
I thought about using it, but the UI design in figma had the button on the left compared to the other sections that has the button on the right, so I decided to go with the UI design.